code/modules/admin/greyscale_modify_menu.dm

```diff
@@ -323,6 +323,7 @@ This is highly likely to cause massive amounts of lag as every object in the gam
 
 /datum/greyscale_modify_menu/proc/DefaultApply()
 	var/atom/atom_target = target
+	atom_target.icon_state = icon_state
```

Comment:
Messing directly with the icon state of something from the GAGS side code is a bad plan. Atoms can tell when they've been given greyscale data via the greyscale procs to know when they need to update if it's important for them to do so. Further, this also allows players to change the icon state themselves through the permission restricted version of the greyscale modification menu.